bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -210,8 +217,22 @@ public BigtableSession(BigtableOptions opts) throws IOException {
     clientInterceptors =
         clientInterceptorsList.toArray(new ClientInterceptor[clientInterceptorsList.size()]);
 
+
     Channel dataChannel = getDataChannelPool();
 
+    Duration waitTimeout = Duration.millis(options.getRetryOptions().getReadPartialRowTimeoutMillis());
```

Comment:
This method is too large, but it's probably worth refactoring it in another PR.  For now, can you please add a `setupWatchdog()` method?

Also does it make sense to have a single watchdog per VM?

Reply:
Cases like Dataflow / PubSub / Bigtable Sink would benefit from a single watchdog.